File: R/02_MLR/code.R
# # Multiple Linear Regression in R using lm()

# # ============ CHANGE THESE ============
# FILENAME <- "data.csv"
# TARGET <- "target"   # dependent variable
# # ======================================

# # --- Load ---
# df <- read.csv(FILENAME)
# cat("Shape:", nrow(df), "x", ncol(df), "\n")
# print(head(df))

# # --- Train/Test Split (80/20) ---
# set.seed(42)
# idx <- sample(1:nrow(df), 0.8 * nrow(df))
# train <- df[idx, ]
# test <- df[-idx, ]
# cat("\nTrain:", nrow(train), "Test:", nrow(test), "\n")

# # --- Fit Model (. means all other columns as predictors) ---
# formula <- as.formula(paste(TARGET, "~ ."))
# model <- lm(formula, data = train)
# print(summary(model))

# # --- Predict ---
# y_pred <- predict(model, newdata = test)
# y_actual <- test[[TARGET]]

# # --- Evaluate ---
# rmse <- sqrt(mean((y_actual - y_pred)^2))
# ss_res <- sum((y_actual - y_pred)^2)
# ss_tot <- sum((y_actual - mean(y_actual))^2)
# r_squared <- 1 - ss_res / ss_tot

# cat("\nR-squared:", round(r_squared, 4), "\n")
# cat("RMSE:", round(rmse, 4), "\n")

#   cat("Actual:", round(y_actual[i], 2), " Predicted:", round(y_pred[i], 2), "\n")
# }

# --- CLEAN & PERFECT MLR ---
df <- read.csv("data.csv")
model <- lm(target ~ ., data = df)
print(summary(model))

# Visualization (Actual vs Predicted)
plot(df$target, predict(model), col="blue", main="Actual vs Predicted")
abline(0, 1, col="red")

File: steyngun_dav/python/mlr/code.py
# Multiple Linear Regression using sklearn
# y = b0 + b1*x1 + b2*x2 + ... + bn*xn

import numpy as np
import pandas as pd
from sklearn.linear_model import LinearRegression
from sklearn.model_selection import train_test_split
from sklearn.metrics import mean_squared_error, r2_score

# ============ CHANGE THESE ============
FILENAME = "data.csv"
X_COLS = ["col1", "col2", "col3"]  # independent variable columns
Y_COL = "target"                    # dependent variable column
# ======================================

# --- Load Data ---
df = pd.read_csv(FILENAME)

X = df[X_COLS]
y = df[Y_COL]

# --- Split ---
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)
print(f"\nTrain: {len(X_train)}, Test: {len(X_test)}")

# --- Train ---
model = LinearRegression()
model.fit(X_train, y_train)
# --- Evaluation & Results ---
y_pred = model.predict(X_test)

print("\n--- MODEL PERFORMANCE ---")
print(f"Intercept: {model.intercept_:.2f}")
print(f"Coefficients: {dict(zip(X_COLS, model.coef_.round(2)))}") # Names & Values in 1 line
print(f"R-squared: {r2_score(y_test, y_pred):.2f}")
print(f"RMSE: {np.sqrt(mean_squared_error(y_test, y_pred)):.2f}")

# Optional: Ultra-short comparison
print("\nActual vs Predicted (Test Set):")
print(pd.DataFrame({'Actual': y_test, 'Predicted': y_pred}).head())

print(dir(model))

dir(model)
